openhands_cli/acp_impl/agent.py

```diff
@@ -53,34 +68,131 @@
 logger = logging.getLogger(__name__)
 
 
+def get_available_modes() -> list[SessionMode]:
+    """Get list of available session modes.
+
+    Returns:
+        List of SessionMode objects representing available modes
+    """
+    return [
+        SessionMode(
+            id="always-ask",
+            name="Always Ask Mode",
+            description="Request permission before making any changes",
+        ),
+        SessionMode(
+            id="always-approve",
+            name="Always Approve Mode",
+            description="Automatically approve all actions without asking",
+        ),
+        SessionMode(
+            id="llm-approve",
+            name="LLM Approve Mode",
+            description=(
+                "Use LLM security analyzer to auto-approve safe actions. "
+                "Only ask for permission on high risk actions."
+            ),
+        ),
+    ]
+
+
+def get_session_mode_state(current_mode: ConfirmationMode) -> SessionModeState:
+    """Get the session mode state for a given confirmation mode.
+
+    Args:
+        current_mode: The current confirmation mode
+
+    Returns:
+        SessionModeState with available modes and current mode
+    """
+    return SessionModeState(
+        current_mode_id=current_mode,
+        available_modes=get_available_modes(),
+    )
+
+
 class OpenHandsACPAgent(ACPAgent):
     """OpenHands Agent Client Protocol implementation."""
 
-    def __init__(self, conn: AgentSideConnection):
+    def __init__(self, conn: Client, initial_confirmation_mode: ConfirmationMode):
         """Initialize the OpenHands ACP agent.
 
         Args:
             conn: ACP connection for sending notifications
+            initial_confirmation_mode: Default confirmation mode for new sessions
         """
         self._conn = conn
         # Cache of active conversations to preserve state (pause, confirmation, etc.)
         # across multiple operations on the same session
-        self._active_sessions: dict[str, BaseConversation] = {}
+        self._active_sessions: dict[str, LocalConversation] = {}
         # Track running tasks for each session to ensure proper cleanup on cancel
         self._running_tasks: dict[str, asyncio.Task] = {}
+        # Track confirmation mode state per session
+        self._confirmation_mode: dict[str, ConfirmationMode] = {}
```

Comment:
It seems like we're still maintaining/coordinating  a separate state for confirmation mode when it could be accessed via `self._active_sessions[convo_id].state.confirmation_policy`

we could replace this dict state with a util to extract the value directly from the conversation object instead